tests/js/basic-tests.js

```diff
@@ -0,0 +1,284 @@
+                    type: "test",
+                    sequence: [
+                        {
+                            func: "{noDataRequest}.send",
+                            args: [{}]
+                        },
+                        {
+                            listener: "gpii.test.couchdb.checkResponse",
+                            event:    "{noDataRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{noDataRequest}.nativeResponse", "{arguments}.0", 200, "{testCaseHolder}.options.expected.noData"]
+                        }
+                    ]
+                },
+                {
+                    name: "Testing reading a single record from the 'sample' database.",
+                    type: "test",
+                    sequence: [
+                        {
+                            func: "{readRequest}.send",
+                            args: [{}]
+                        },
+                        {
+                            listener: "gpii.test.couchdb.checkResponse",
+                            event:    "{readRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{readRequest}.nativeResponse", "{arguments}.0", 200, "{testCaseHolder}.options.expected.read"]
+                        }
+                    ]
+                },
+                {
+                    name: "Confirm that supplemental data was loaded for the 'sample' database.",
+                    type: "test",
+                    sequence: [
+                        {
+                            func: "{supplementalReadRequest}.send",
+                            args: [{}]
+                        },
+                        {
+                            listener: "gpii.test.couchdb.checkResponse",
+                            event:    "{supplementalReadRequest}.events.onComplete",
+                            //        (response, body, expectedStatus, expectedBody)
+                            args:     ["{supplementalReadRequest}.nativeResponse", "{arguments}.0", 200, "{testCaseHolder}.options.expected.supplementalRead"]
+                        }
+                    ]
+                },
+                {
+                    name: "Testing deleting a single record from the 'sample' database.",
+                    type: "test",
+                    sequence: [
+                        // The record should exist before we delete it.
+                        {
+                            func: "{preDeleteRequest}.send",
+                            args: [{}]
+                        },
+                        // confirm that the record exists now and delete the latest revision.
+                        {
+                            listener: "gpii.tests.couchdb.basic.checkRecordAndStartDelete",
```

Comment:
Is it possible to handle preDelete (the fetch of _rev) internally and silently?

Reply:
This behaviour is actually a part of CouchDB, just like you're required to send a revision to update an entry.   express-pouchdb is not entirely faithful to the way in which CouchDB works, which is part of the reason I created this package.  We should be able to confidently know that our code that performs DELETEs will work against a real CouchDB instance.

We shouldn't make the harness handle DELETEs any differently than CouchDB does, but I understand that this double-request process can be cumbersome.  I haven't found anyone actually performing real deletes in universal or my own work, but once someone needs to do so,  we should create a variation on the CouchDB dataSource that can either make use of the DELETE method, or that can pass the special `_deleted` field as part of an update request.